perfkitbenchmarker/disk_iops_to_capacity.py

```diff
@@ -0,0 +1,259 @@
+      raise InvalidProviderError('Provider given is not supported by '
+                                 'storage_utility.')
+
+  def _ValidateIOPS(self):
+    """Validate IOPS to be within valid limits, throw exception if invalid.
+
+    If IOPS parameter is less than 1 or greater than provider maximum IOPS
+    throw InvalidIOPSError.
+
+    Raises:
+      InvalidIOPSError: Invalid IOPS parameter given.
+    """
+    if (self._iops < 1 or
+        self._iops > CLOUD_PROVIDERS_INFO[self._provider][MAX_IOPS]):
+      raise InvalidIOPSError()
+
+  def _PopulateConfigs(self):
+    """Populate Storage Configurations."""
+    self._SetSize()
+    self._SetCPUCount()
+    self._SetNumberDisks()
+
+  def PrintConfigs(self):
+    """Print out necessary configs."""
+    vm_config = ('For {} IOPS using {}, the following is required:\n\tStorage '
+                 'Size (GB): {}\n\tCPU Count: {}\n\tNumber of '
+                 'Disks: {}').format(self._iops,
+                                     self._provider.upper(), self._size,
+                                     self._cpu_count, self._number_disks)
+    print vm_config
+
+  def _SetSize(self):
+    """Set minimum size (GB) necessary to achieve _iops level.
+    Rating performance levels as of May 2017, sources found below.
+    GCP: ratings from https://cloud.google.com/compute/docs/disks/. Storage can
+          go as high as 64TB per disk but IOPS maxes out at 30,000 iops/disk
+          or (1000GB).
+    AWS: ratings from
+          http://docs.aws.amazon.com/AWSEC2/latest/
+          UserGuide/EBSVolumeTypes.html#EBSVolumeTypes_gp2. Storage can go as
+          high as 16TiB per disk but IOPS maxes out at 10000 IOPS/volume size.
+          Reference gives volume size in GiB so converted to GB below.
+    """
+    if self._provider == GCP:
+      self._size = min(int(math.ceil(self._iops / 30.0)), 30000 / 30)
+    elif self._provider == AWS:
+      value = self._iops
+      value = numpy.array(value)
+      self._size = int(
+          numpy.piecewise(value, [[value <= 100], [(value > 100) & (
+              value <= 9999)], [value > 9999]], [
+                  lambda x: int(math.ceil(1.07374)),
+                  lambda x: int(math.ceil(3 * value)),
+                  lambda x: int(math.ceil(3579.855))]))
+
+  def GetSize(self):
+    """Return storage size.
+    Returns:
+      __size: Storage size (GB).
+    """
+    return self._size
+
+  def _SetCPUCount(self):
+    """Set cpu count.
+
+    GCP: ratings from
+    https://cloud.google.com/compute/docs/disks/performance#ssd-pd-performance
+    AWS: ratings from https://aws.amazon.com/ebs/details/
+    """
+    if self._provider == GCP:
+      value = self._iops
+      self._cpu_count = int(
+          numpy.piecewise(value, [[value <= 15000], [
+              (value > 15000) & (value <= 25000)
+          ], [value > 25000]], [lambda x: 1, lambda x: 16, lambda x: 32]))
+    elif self._provider == AWS:
+      self._cpu_count = 1
```

Comment:
For AWS, to achieve good performance on EBS, one needs to use an EBS-optimized VM instance, and the smallest VM instance that can be EBS optimized is *.large VM types (e.g., c4.large), those comes with 2 cores. See link below.

http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSOptimized.html

So we need to change the cpu_count here for AWS to 2, and supply the above link to the doc of this function.